src/dmd/expressionsem.d

```diff
@@ -5269,7 +5269,12 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
                     // Perturb the name mangling so that the symbols can co-exist
                     // instead of colliding
                     s.localNum = cast(ushort)(originalSymbol.localNum + 1);
-                    assert(s.localNum);         // 65535 should be enough for anyone
+                    // 65535 should be enough for anyone
+                    if (!s.localNum)
+                    {
+                        e.error("too many symbols with name `%s`", s.ident.toChars());
```

Comment:
"too many" leaves the user dangling wondering how many is too many. A better message would be along the lines of : "more than 65534 symbols generated".